### ISSUE TITLE
Allow to set a time difference `PULSE_DELAY` with respect to simulation start at which the maximum of a laser profiles enters the simulation box

### ISSUE THREAD

**Issue by steindev** [ComputationalRadiationPhysics/picongpu#4323]
See discussion in PR #4302.

**Comment by finnolec** [comment]
Is it maybe better, to set `PULSE_DELAY` to the time (in SI units), when the laser pulse is in its focus position? At least for probing simulations this would require less adjustments, when one e.g. resizes the simulation box. But I can see, how this can be weird to define for the non-focused lasers, [although they also have](https://github.com/ComputationalRadiationPhysics/picongpu/blob/dev/include/picongpu/fields/incidentField/profiles/BaseParam.def#L120) the `FOCUS_POSITION_X_SI` etc attributes. 

Edit: Nevermind, this would also lead to weird simulation set-ups for e.g. regular drive lasers in LWFAs.

Edit 2: On the other hand, the implementation of `PULSE_DELAY` as the time when the laser maximum enters the simulation box can become unintuitive for tilted laser pulses or for lasers, that enter the the simulation box with non-orthogonal angles.

**Comment by steindev** [comment]
Agreed, we have to think thoroughly to find a definition appropriate for all our use cases.

**Comment by psychocoderHPC** [comment]
discussed in dev meeting: Maybe it is not useful to have only a single value`PULSE_DELAY`, maybe we need a functor to handle different use cases where the delay has a different meaning.
Independent of the outcome we should keep the PICMI standard in mind that we not build something which is hard to use or implement with PICMI.
